### ISSUE TITLE
only-arrow-functions rule should ignore functions that use the 'arguments' object

### ISSUE THREAD

**Issue by Zyphrax** [palantir/tslint#1540]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.1.0-dev.20160906`
- __Running TSLint via__: Visual Studio Code

#### TypeScript code being linted

```ts
// Redirect logging to native, WKWebView captures the alert and logs the message
console.log = function () { alert(["info", Array.prototype.slice.call(arguments)]); };
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:latest",
  "rules": {
    "max-line-length": [false],
    "member-ordering": [true, { "order": "fields-first" }],
    "no-namespace": false,
    "no-console": [],
    "object-literal-sort-keys": false,
    "ordered-imports": false,
    "trailing-comma": "never",
    "triple-equals": false
  }
}
```

#### Actual behavior

TSLint reports that non-arrow functions are forbidden. Which is a good default. However when you want to use the `arguments` keyword in typescript you have to use the classic javascript function notation.

The following code:
```ts
console.debug = () => { alert(["debug", Array.prototype.slice.call(arguments)]); };
```

Will result in the following error:

> error TS2496: The 'arguments' object cannot be referenced in an arrow function in ES3 and ES5. Consider using a standard function expression.

#### Expected behavior

TSLint should not apply the only-arrow-functions rule when the function refers to `arguments`. At least not if the project targets ES5 or lower.

**Comment by JKillian** [comment]
Instead of using `arguments`, could you do something more in line with ES6 like this?

```
console.debug = (...args) => { alert(["debug", Array.prototype.slice.call(args)]); };
```

**Comment by Zyphrax** [comment]
@JKillian Great suggestion, thank you!

**Comment by JKillian** [comment]
Agreed, that could be a good rule. I'm going to close this issue though as we have a better solution than changing the `only-arrow-functions` rule
